socket.go

```diff
@@ -384,10 +384,16 @@ func (sck *socket) connReaper() {
 			return
 		}
 
-		for _, c := range sck.closedConns {
+		// Clone the known closed connections to avoid data race
+		// and remove those under reaper unlocked.
+		// That would be resoling deadlock from #149 simpler way.
+		cc := append([]*Conn{}, sck.closedConns...) // clone
+		sck.closedConns = nil
```

Comment:
should we reuse the slice ?
```suggestion
		sck.closedConns = sck.closedConns[:0]
```

I don't know.